src/main/java/hudson/plugins/analysis/graph/CategoryBuildResultGraph.java

```diff
@@ -239,12 +243,21 @@ protected JFreeChart createChart(final GraphConfiguration configuration, final R
         return valuesPerBuild;
     }
 
-    private boolean passesFilteringByParameter(final AbstractBuild<?, ?> build, final String parameterName, final String parameterValue) {
+    private boolean passesFilteringByParameter(final Run<?, ?> build, final String parameterName, final String parameterValue) {
         if (parameterName == null) {
             return true;
         }
 
-        Map<String, String> variables = build.getBuildVariables();
+        Map<String, String> variables;
+        if (build instanceof AbstractBuild) {
+            variables = ((AbstractBuild<?, ?>) build).getBuildVariables();
+        }
+        else {
+            // There is no comparable method for Run. This means that this feature (using parameters for
+            // result graph) will not be available for other than AbstractBuild extending classes (basically
+            // all except Workflow builds).
+            variables = null;
```

Comment:
There is a comment above about this choice. Returning `true` does not make sense since we have no equivalent to `AbstractBuild.getBuildVariables()` for `Run`, so keeping the current logic (variables == null then return false).